### ISSUE TITLE
Add image from Wikimedia Commons via Wikidata

### ISSUE THREAD

**Issue by b2d3c** [sapa/performing-arts-ch-templates#328]
We would like to try to add automaticaly Image from Wikimedia Commons on our Portal.

For example : Fred Erismann

Our Entity : http://data.performing-arts.ch/a/cc75208a-9788-4669-acda-2ada09254fde (with value owl:sameAs on Wikidata Entity)

Wikidata Entity : http://www.wikidata.org/entity/Q102180314 with link (via property http://www.wikidata.org/entity/P18) on one image on Wikimedia Commons (https://www.wikidata.org/wiki/Q102180314#P18)

With all this information, could we generate this code : 

`<p><a href="https://commons.wikimedia.org/wiki/File:Theaterfotograf_Fred_Erismann_bei_der_Arbeit.png#/media/File:Theaterfotograf_Fred_Erismann_bei_der_Arbeit.png"><img src="https://upload.wikimedia.org/wikipedia/commons/b/b2/Theaterfotograf_Fred_Erismann_bei_der_Arbeit.png" alt="Theaterfotograf Fred Erismann bei der Arbeit.png" width="663" height="827"></a><br>Par Margrit Erismann (Ehefrau des Fred Erismann) — Privatsammlung Fred Erismann, <a href="https://creativecommons.org/licenses/by-sa/4.0" title="Creative Commons Attribution-Share Alike 4.0">CC BY-SA 4.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=96211145">Lien</a></p>`

or a light version of it, and publish it automaticaly on our Entity?

See also : #450

**Comment by b2d3c** [comment]
@fkraeutli do you think, it's complex to do this?

**Comment by fkraeutli** [comment]
Hi Baptiste! We have a pipeline that does this, but I'd have to check how adaptable it is to your platform. What we also need to keep in mind are the image credits. Unfortunately they are not on Wikidata, but on Wikimedia, which uses a different API. We also have a pipeline to retrieve this data, but again, I have to check if we can adapt this part too.

**Comment by fkraeutli** [comment]
What could be done more easily is to query for the image on the fly. The downside of this is that when Wikidata is unreachable it might produce an error or be unavailable. But since having an image is not critical, this might still be a good option.

**Comment by b2d3c** [comment]
On the fly is perfekt for us.

**Comment by olyanicolaeva** [comment]
To query for Wikidata images we can use the following query:
```
PREFIX crm: <http://www.cidoc-crm.org/cidoc-crm/>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
prefix wdt: <http://www.wikidata.org/prop/direct/>
SELECT ?value ?subject WHERE {?subject a crm:E21_Person; owl:sameAs ?wikiId . 
SERVICE <https://query.wikidata.org/sparql> {
    ?wikiId wdt:P18 ?value . 
FILTER (STRSTARTS(str(?wikiId), 'http://www.wikidata.org/entity/'))
}}
```
Most economic solution is to add the query into **preferredThumbnails** configurations, since the template is using `knowledgeGraphBar__thumbnail` component: [template](https://github.com/sapa/performing-arts-ch-templates/blob/74f9a2622b772b4ee2049d03df1c26f1fea5a7e2/data/templates/http%253A%252F%252Fpage-module.performing-arts.ch%252FKnowledgeGraphBar.html)

Two steps are required:
-  add `wdt` namespace to the platform configutarion
- modify the query to fit the standards of preferredThumbnails ([Documentation](https://help.metaphacts.com/resource/?uri=Help%3ABasicSystemConfiguration)) like this:
```
{?subject a crm:E21_Person; owl:sameAs ?wikiId . 
SERVICE <https://query.wikidata.org/sparql> {
    ?wikiId wdt:P18 ?value . 
FILTER (STRSTARTS(str(?wikiId), 'http://www.wikidata.org/entity/'))
}}
```
- additional CSS alignment to scale the image correctly

**Comment by b2d3c** [comment]
For me, seems perfekt. Could we make a test in dev?

**Comment by olyanicolaeva** [comment]
@b2d3c yes, will do

**Comment by fkraeutli** [comment]
Deployed to dev

**Comment by b2d3c** [comment]
It's Great ! 

Two asks : 
1. Could the size of the icon "person" (in case of no image in wikidata) be as big as it was before the implentation
 * ex : 
![image](https://user-images.githubusercontent.com/8588224/224319265-8942f35a-30af-4f5c-b588-ac094cdfb56d.png)
![image](https://user-images.githubusercontent.com/8588224/224319383-09ccac58-ab49-45c7-b897-7fc6e0c8eb0e.png)
2. Cloud we add the mention "Wikimedia Commons" (or icon) under the image with the link on the page of the file in Commons?

**Comment by olyanicolaeva** [comment]
@b2d3c yes, we can do both!

**Comment by b2d3c** [comment]
hello @olyanicolaeva , we have seen it's already on prod deployed : http://data.performing-arts.ch/a/49187985-04f9-4280-be72-dc316d9c1e01

Please could you insert the link on Wikimedia commons under the Picture, Thanks.

**Comment by b2d3c** [comment]
Exemple : 
![image](https://github.com/sapa/performing-arts-ch-templates/assets/8588224/05a1f6b1-53f5-4d96-8bae-ca349f4479b9)

**Comment by olyanicolaeva** [comment]
hi @b2d3c , yes, we can do it this week

**Comment by olyanicolaeva** [comment]
@b2d3c there are problems with the css code in this template, so I had to change the position of the caption
<img width="1518" alt="Screenshot 2023-05-23 at 12 21 46" src="https://github.com/sapa/performing-arts-ch-templates/assets/86773892/1e5ae4ec-8dab-4ede-aba2-2337d001c876">
It will look like this (deployed to dev). But there is another problem with the initial design of the template, so I will refactor it this week (so that the icon is displayed correctly for persons without images)

**Comment by b2d3c** [comment]
Dear @olyanicolaeva , 
It's all right! It's seem perfekt.

**Comment by b2d3c** [comment]
Dear @olyanicolaeva, if it's too complex and take to many time, we can with no problem think again together about it.

We have already new ideas how to make it. Perhaps make we a short call to discuss it?

Poke: @fkraeutli

**Comment by olyanicolaeva** [comment]
@b2d3c I will let you know if I can fix it today and we can discuss the details if needed.

**Comment by b2d3c** [comment]
See also : #450

**Comment by fkraeutli** [comment]
merged into #450
